apteryxd.c

```diff
@@ -1894,6 +1894,36 @@ collect_provided_paths_query(GNode *query)
     return matches;
 }
 
+static gboolean _refresh_paths (GNode *node, gpointer data)
+{
+    char *path = NULL;
+
+    /* Handle wildcards */
+    if (g_strcmp0 (node->data, "*") == 0)
+    {
+        /* Match everything from here down and go no further */
+        // TODO handle mid path wildcards that might not match further down
+        _node_to_path (node->parent, &path);
+        refreshers_traverse (path, cb_all);
+        free (path);
+        return FALSE;
+    }
+
+    /* Handle direct matches */
+    if (g_node_n_children (node) == 1 && (!node->children->data || g_node_n_children (node->children) == 0))
+    {
+        /* Match this exactly and go no further */
+        _node_to_path (node, &path);
+        refreshers_traverse (path, cb_all);
```

Comment:
I think this can just be call_refreshers without needing the whole traverse - we're at the end anyway